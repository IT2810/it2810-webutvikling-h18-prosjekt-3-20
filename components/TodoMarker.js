import React from 'react';
import PropTypes from 'prop-types';
import { Marker } from 'react-native-maps';

const TodoMarker = ({ todo }) => {
  const { name, coordinates } = todo;

  // Map shorthand keys lat and lon,
  // to latitude and longitude
  const coord = {
    latitude: coordinates.lat,
    longitude: coordinates.lon,
  };

  return <Marker title={name} coordinate={coord}/>;
};
TodoMarker.propTypes = {
  todo: PropTypes.shape({
    name: PropTypes.string.isRequired,
    coordinates: PropTypes.object.isRequired,
  }).isRequired,
};

export default TodoMarker;
