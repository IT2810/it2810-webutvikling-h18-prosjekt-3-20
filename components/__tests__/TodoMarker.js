import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import TodoMarker from '../TodoMarker';

const mockedTodo = {
  completed: false,
  coordinates: {
    latitude: 63.41531291552631,
    longitude: 10.405357924043026,
  },
  date: '2018-10-20',
  distance: 7,
  id: 'Gkrkt2018-10-20',
  name: 'Gkrkt',
};

/**
 * TodoMarker is only a wrapper component around react-native-maps's Marker,
 * which makes the tests necessary to run quite limited.
 * Input validation is done by PropTypes and the actual rendering and user
 * interaction is handled by Marker. This is why
 * */
describe('<TodoMarker/>', () => {
  it('should render without failing', () => {
    const tree = renderer.create(<TodoMarker todo={mockedTodo}/>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
