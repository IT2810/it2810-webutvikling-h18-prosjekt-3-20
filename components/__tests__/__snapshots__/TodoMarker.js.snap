// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoMarker/> should render without failing 1`] = `
<AIRMapMarker
  coordinate={
    Object {
      "latitude": 63.41531291552631,
      "longitude": 10.405357924043026,
    }
  }
  description="Due: 2018-10-20"
  image={undefined}
  onPress={[Function]}
  pinColor="#FF5555"
  stopPropagation={false}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "position": "absolute",
      },
      undefined,
    ]
  }
  title="Gkrkt"
/>
`;
